### PR TITLE
fix: Correct Guid Serialization

### PR DESCRIPTION
**Pull request by paulhazen**
Because the eos sdk needs to have any Guid value be input as a string with no dashes, this PR introduces a custom GuidConverter that performs that task automatically.

This prevents the native code from needing to do the manipulation.

The associated JSON files were updated as well.

Since the way the converter works seems like a more straightforward and non-intrusive means of accomplishing what GuidExtensions used to do, that was also removed in favor of using the `.ToString("N").ToLowerInvariant()` method.

### CODE REVIEW

File: com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            var stringValue = reader.Value?.ToString();
+
+            if (string.IsNullOrEmpty(stringValue))
+            {
+                return objectType == typeof(Guid?) ? (Guid?)null : Guid.Empty;
+            }
+
+            if (!Guid.TryParseExact(stringValue, "N", out Guid value))
+            {
+                value = Guid.Parse(stringValue);
```
Comment (paulhazen) [inline com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs]: Might be a good idea to use `TryParse`?

Comment (paulhazen) [reply com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs]: This has been resolved - the behavior is to use TryParse, and if it fails to set the returning value to be `Guid.Empty`. This behavior is consistent with other classes that implement the `JsonConverter` class.

File: com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+*/
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Newtonsoft.Json;
+    using System;
+
+    public class GuidConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
```
Comment (WispyMouse) [inline com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:30]: I noticed that `JsonConverter` uses `object?` for the parameter `value`, `existingValue`, and the return type for `ReadJson`. Should we match that, or does the `?` not work with our version of C#?

File: com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            if (value is Guid guid)
```
Comment (WispyMouse) [inline com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:32]: Will this work if `value` is `Guid?`, or only if the value type is exactly `Guid`?

Comment (paulhazen) [reply com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:32]: It's not possible for value to be `Guid?`

File: com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs (hunk 2 above)
Comment (paulhazen) [reply com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:30]: `?` works - but this is only applied to field members that are of type `Guid`, not any that are of type `Guid?`

**Review by WispyMouse — APPROVED** [review]
Left some thoughts, but otherwise approved.

File: com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs (hunk 3 above)
Comment (andrew-hirata-playeveryware) [reply com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:32]: Perhaps I don't understand this JSON writer well enough, but why is possible to Read JSON where the type is a `Guid?` but not write one?

Comment (paulhazen) [reply com.playeveryware.eos/Runtime/Core/Config/JsonConverter/GuidConverter.cs:32]: It is possible - it's just that we don't have a need to implement it for `Guid?`

We _could_ add it for future-proofing - but that would be adding functionality that isn't currently needed, which seemed like a bad thing to do. Support for `Guid?` would be easy to add though - if you'd like it added.

**Review by andrew-hirata-playeveryware — APPROVED** [review]

**Pull request merged**